supportingFunctions.R

```diff
@@ -0,0 +1,304 @@
+##R Final Project
+# Written by Isaiah Murrell-Thomas and Bethany Oceguera 
+
+## IMPORTANT INFORMATION
+# Before running this code, please make sure that you are in the directory containing all relevant files 
+
+##FUNCTION 1
+# Function that converts all files in a directory with space- or 
+#   tab-delimited data (.txt) into comma-separated value files.
+
+csvConvert <- function(dir){
+  setwd(dir)
+  filesToConvert <- list.files(pattern = ".txt") 
+  for (i in 1:length(filesToConvert)){
+    input <- filesToConvert[i]  
+    output <- paste0(gsub("\\.txt$", "", input), ".csv")
+    data <- read.table(input, sep="",header=TRUE)
+    write.table(data, file=output, sep=",", col.names=TRUE, row.names=FALSE)
+    file.remove(input)  
+  }
+  print("Files have been converted to .csv")
+}
+
+#USAGE 
+# Define and run variable "dir" as the directory path that holds the files to be converted
+# Run "csvConvert(dir)"
+
+#EXAMPLE: convert and remove .txt files in "countryY" directory
+# dir <- "/Users/bethoceguera/Documents/PhD Year 1/First Semester/Introduction to Biocomputing/Rproject/countryY" 
+# csvConvert(dir)
+
+
+##FUNCTION 2
+# Function that compiles data from all .csv files in a directory 
+# into a single .csv file. The compiled data has the original twelve columns 
+# from daily data sheets, but also country and dayofYear columns. 
+# The user should be able to choose whether they want to remove rows with NA’s 
+# in any columns, include NAs in the compiled data but be warned of their presence, or include NAs in the compiled data without a warning
+
+csvMerge<-function(dir,naChoice){
+  myPath <- list.files(path = dir, recursive = TRUE, pattern = "*.csv", full.names = TRUE)
+  dirName <-dirname(path= myPath)
+  dayofYear <-'dayofYear'
+  country<-'country'
+  header<-read.csv(myPath[1],header = TRUE, stringsAsFactors = FALSE) 
+  for (i in seq_along(myPath)){
+    if (naChoice=="1"){
```

Comment:
You can merge the data and add the columns before choosing to remove NA values. By doing this, you don't need to repeat the codes of adding the columns within every conditions.